docs/criusupport.md

```diff
@@ -0,0 +1,99 @@
+* License, Version 2.0 which accompanies this distribution and
+* is available at https://www.apache.org/licenses/LICENSE-2.0.
+*
+* This Source Code may also be made available under the
+* following Secondary Licenses when the conditions for such
+* availability set forth in the Eclipse Public License, v. 2.0
+* are satisfied: GNU General Public License, version 2 with
+* the GNU Classpath Exception [1] and GNU General Public
+* License, version 2 with the OpenJDK Assembly Exception [2].
+*
+* [1] https://www.gnu.org/software/classpath/license.html
+* [2] https://openjdk.org/legal/assembly-exception.html
+*
+* SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH
+* Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+-->
+
+# CRIU support
+
+**(Linux&reg; x86, Linux on POWER&reg; (Little Endian), and Linux on IBM Z&reg; only)**
+
+:fontawesome-solid-pencil:{: .note aria-hidden="true"} **Note:** Support for the Checkpoint/Restore In Userspace (CRIU) tool is currently provided as a technical preview in container environments. CRIU support is available for CRIU version that is packaged with the Semeru package. This preview is supported for use in production environments, however, all APIs and command-line options are subject to change.
+
+In OpenJ9, the CRIU support includes an API that you can use to stop the VM at a checkpoint, save its state, and then run the VM from the point where it was stopped. The period of time between when the VM starts and when the application takes a checkpoint is referred to as the *checkpoint phase*. The application writes the VM state that was saved at the checkpoint to one or more image files. The saved state includes memory pages, methods, file systems, processes, and network connections. You can restore the VM from these files multiple times from the same point and in multiple environments.
+
+Because the checkpoint image files have the live state of the VM that is used to restore the VM in different systems, they should not contain cryptographically-sensitive security data. If the image files contain sensitive security data, the security components are vulnerable to exploitation even if you don't move the image files between systems. The CRIU technical preview introduces the `CRIUSECProvider` security provider, which provides the following limited set of security services: `MessageDigest` and `SecureRandom`. When you enable CRIU support, all existing security providers are removed from the security provider list during the checkpoint phase and `CRIUSECProvider` is added. When you restore the VM in the nonportable restore mode ([`-XX:+CRIURestoreNonPortableMode`](xxcriurestorenonportablemode.md)), `CRIUSECProvider` is removed from the security provider list and the previous security providers are added back again.
+
+When the VM starts from the saved point instead of the beginning, the VM startup time improves and hence results in better VM performance.
```

Comment:
"and hence results in better performance" seems superfluous

Reply:
Would it not result in better VM performance or does "...the VM startup time improves" imply that VM performance is better as a result? If both mean same I will delete it. Please let me know. Thanks!

Reply:
Yes the main aspect that is supposed to improve from a performance viewpoint is that startup time will be better. The rest of the metrics are mostly neutral from this feature (we have seen some small footprint gains but those need to be understood better and they are minor anyway).

Reply:
So can I leave "and hence results in better performance" sentence as-is?

Reply:
No, I am saying that the only tangible benefit is to startup time and so ""and hence results in better performance" should be deleted.

Reply:
Deleted.